lib/YuiRestClient/Widget/SingleItemSelector.pm

```diff
@@ -0,0 +1,32 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YuiRestClient::Widget::SingleItemSelector;
+
+use strict;
+use warnings;
+
+use parent 'YuiRestClient::Widget::Base';
+use YuiRestClient::Action;
+
+sub select {
+    my ($self, $item) = @_;
+    return $self->action(action => YuiRestClient::Action::YUI_SELECT, value => $item);
+}
+
+sub selected {
+    my ($self)     = @_;
+    my $items      = $self->property('items');
+    my ($selected) = map { $_->{label} } grep { $_->{selected} } @{$items};
+    return $selected || 'none';
```

Comment:
I would try to avoid magic string. I see your motivation to make it transparent from the test data, but I believe this is wrong place for this string. If we really want to have it, it should be also defined as a constant somewhere, where it's visible. I would suggest that we do mapping of `none` to undef or empty array on the test level. I doubt that we will ever have item called `none`, but if we face it, there is no way we can differentiate between nothing is selected and `none` being selected in the list.

Reply:
I will use yaml syntax for null value then to be more strict.